### ISSUE TITLE
Support for Swift 4

### ISSUE THREAD

**Issue by nab0y4enko** [airmap/AirMapSDK-Swift#23]
```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '10.0'
use_frameworks!

def shared_pods
    # Network and mapping
    pod 'Alamofire'
    pod 'AlamofireImage'
    pod 'ObjectMapper'

    # Chat
    pod 'SendBirdSDK'

    # Other
    pod 'SwiftyUserDefaults'
    pod 'DZNEmptyDataSet'
    pod 'PrettyExtensionsKit'

    # AWS
    pod 'AWSS3'
end

target 'MyApp' do
    # Shared
    shared_pods

    # Network and mapping
    pod 'AlamofireNetworkActivityIndicator'

    # UI
    pod 'SVProgressHUD'
    pod 'SideMenu'
    pod 'ISHPullUp'
    pod 'PrettyFloatingMenuView'
    pod 'TTTAttributedLabel', :git => 'https://github.com/TTTAttributedLabel/TTTAttributedLabel.git', :branch => 'master'
    pod 'ElValidator'
    pod 'SlackTextViewController', :git => 'https://github.com/slackhq/SlackTextViewController.git', :branch => 'master' # fixed bottom UITextView for iOS 11
    pod 'BadgeSwift'
    pod 'RKTagsView'
    pod 'PMAlertController'

    # Autorization
    pod 'Lock'

    # Map
    pod 'CryptoSwift', :git => 'https://github.com/krzyzanowskim/CryptoSwift/', :branch => 'develop'
    pod 'AirMapSDK/Core', :git => 'https://github.com/airmap/AirMapSDK-Swift/', :branch => 'chore/xc9'
    pod 'AirMapSDK/UI', :git => 'https://github.com/airmap/AirMapSDK-Swift/', :branch => 'chore/xc9'
    pod 'AirMapSDK/Traffic', :git => 'https://github.com/airmap/AirMapSDK-Swift/', :branch => 'chore/xc9'
    pod 'GooglePlaces'

    # Analytics
    pod 'Fabric'
    pod 'Crashlytics'
    pod 'Instabug'

    # Other
    pod 'KeychainSwift'
    pod 'PhoneNumberKit'
    pod 'HotlineSDK'
    pod 'PrettyMaidenheadLocator'   
    pod 'Base64'
    pod 'ColorThiefSwift'
    pod 'TPKeyboardAvoiding'

    target 'MyAppTests' do
        inherit! :complete
    end
end

target 'MyAppChatShareExtension' do
    # Shared
    shared_pods
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        if ['AirMapSDK', 'PhoneNumberKit', 'Lock', 'RxSwift', 'RxSwiftExt', 'RxCocoa', 'RxDataSources'].include? target.name
            target.build_configurations.each do |config|
                config.build_settings['SWIFT_VERSION'] = '3.2'
            end
        end
    end
end

```

Of all the libraries used, only AirMap(with dependencies) does not have the support of Swift 4.
> 'AirMapSDK', 'PhoneNumberKit', 'Lock', 'RxSwift', 'RxSwiftExt', 'RxCocoa', 'RxDataSources'

Please take care of your SDK.

**Comment by adolfo** [comment]
Thanks for the feedback. Looks like you found the workaround for setting the Swift version in your Podfile.
```
post_install do |installer|
    installer.pods_project.targets.each do |target|
        if ['AirMapSDK', 'PhoneNumberKit', 'Lock', 'RxSwift', 'RxSwiftExt', 'RxCocoa', 'RxDataSources'].include? target.name
            target.build_configurations.each do |config|
                config.build_settings['SWIFT_VERSION'] = '3.2'
            end
        end
    end
end
```

We are working on a major new version of this SDK and will provide Swift 4 support there.

**Comment by adolfo** [comment]
@nab0y4enko Swift 4 support is now available via a `2.0.0.beta.1` release tag. Please add the following to your Podfile to use it:

```ruby
...
pod 'AirMapSDK', :git => 'https://github.com/airmap/AirMapSDK-Swift', :tag => '2.0.0.beta.1'

post_install do |installer|
    # List of Pods to use as Swift 3.2
    swift3_targets = ['Lock', 'SwiftMQTT'] 
    
    installer.pods_project.targets.each do |target|
        if swift3_targets.include? target.name
            target.build_configurations.each do |config|
                config.build_settings['SWIFT_VERSION'] = '3.2'
            end
        end
    end
end
...
```

The last two remaining external Swift 3 dependencies should be upgraded soon, so you'll still need that post install script.
